tools/jit/gen_jit_dispatch.py

```diff
@@ -91,15 +91,20 @@ def from_ivalue(arg, value):
     ${args}
 );
 """)
-CALL_TENSOR_OPTIONS = CodeTemplate("""\
+CALL_NAMESPACE_WITH_TENSOR_OPTIONS = CodeTemplate("""\
 const auto options = TensorOptions()
         .dtype(${dtype})
         .layout(${layout})
         .device(${device});
-auto result = torch::${name}(
-    ${args},
-    options
-);
+auto result = torch::${name}(${args_with_tensor_options});
+""")
+CALL_METHOD_WITH_TENSOR_OPTIONS = CodeTemplate("""\
+const auto options = TensorOptions()
+        .dtype(${dtype})
+        .layout(${layout})
+        .device(${device});
+DeviceGuard device_guard(deviceForInputs(stack, ${num_inputs}));
```

Comment:
DeviceGuard is no longer needed here. I have another open PR that removes it from the other place in the file.